File: MyDailyBeat/MyDailyBeat/EVCResourceLinksTableViewController.m
//
//  EVCResourceLinksTableViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 8/30/15.
//  Copyright (c) 2015 eVerveCorp. All rights reserved.
//

#import "EVCResourceLinksTableViewController.h"

@interface EVCResourceLinksTableViewController ()

@end

@implementation EVCResourceLinksTableViewController

- (id) initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil andModuleName: (NSString *) m {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self->module = m;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    path = RES_LINKS;
    resLinks = [NSDictionary dictionaryWithContentsOfFile:path];
    UIEdgeInsets insets = UIEdgeInsetsMake(0, 0, CGRectGetHeight(self.tabBarController.tabBar.frame), 0);
    self.tableView.contentInset = insets;
    self.tableView.scrollIndicatorInsets = insets;
    self.edgesForExtendedLayout = UIRectEdgeAll;
    [self reloadData];
     
    
}

- (void) reloadData {
    if ([module isEqualToString:@"Finance"]) {
        _dataArr = [resLinks objectForKey:@"Finances"];
    } else if ([module isEqualToString:@"FeelingBlue"]) {
        _dataArr = [resLinks objectForKey:@"FeelingBlue"];
    } else if ([module isEqualToString:@"Relationships"]) {
        _dataArr = [resLinks objectForKey:@"Relationships"];
    } else if ([module isEqualToString:@"Jobs"]) {
        _dataArr = [resLinks objectForKey:@"Jobs"];
    } else if ([module isEqualToString:@"Health"]) {
        _dataArr = [resLinks objectForKey:@"Health"];
    } else if ([module isEqualToString:@"Travel"]) {
        _dataArr = [resLinks objectForKey:@"Travel"];
    } else if ([module isEqualToString:@"Volunteering"]) {
        _dataArr = [resLinks objectForKey:@"Volunteering"];
    } else if ([module isEqualToString:@"Shopping"]) {
        _dataArr = [resLinks objectForKey:@"Shopping"];
    }else {
        _dataArr = [[NSArray alloc] init];
    }
    [self.tableView reloadData];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {

    // Return the number of rows in the section.
    return [_dataArr count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CELL"];
    
    if ([[_dataArr objectAtIndex:indexPath.row] isEqualToString:@""]) {
        cell.textLabel.text = @"No Links Found";
    } else {
        cell.textLabel.text = [_dataArr objectAtIndex:indexPath.row];
    }
    
    
    return cell;
}


#pragma mark - Table view delegate

// In a xib-based application, navigation from a table can be handled in -tableView:didSelectRowAtIndexPath:
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [self openURLinBrowser:[_dataArr objectAtIndex:indexPath.row]];
}

- (void) openURLinBrowser: (NSString *) url {
    NSString *fullURL = [NSString stringWithFormat:@"http://www.%@", url];
    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:fullURL]];
    
}


@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCResourceLinksTableViewController.swift

//
//  EVCResourceLinksTableViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 8/30/15.
//  Copyright (c) 2015 eVerveCorp. All rights reserved.
//
import UIKit
import API
import SwiftyJSON

fileprivate let RESOURCE_LINKS: [String: [String]] = [
    "Finance": ["finance.yahoo.com", "money.cnn.com", "bloomberg.com"],
    "FeelingBlue": ["suicidepreventionlifeline.org", "psychcentral.com/helpme.htm", "hopeline.com"],
    "Jobs": ["livecareer.com", "careerservices.princeton.edu"],
    "Health": ["webmd.com", "mayoclinic.org", "cdc.gov", "medicineonline.com", "online-medical-dictionary.org"],
    "Relationships": [],
    "Shopping": [],
    "Travel": ["travelchannel.com", "cnn.com/travel", "forbestravelguide.com", "tripadvisor.com"],
    "Volunteering": ["signup.com", "agingnetworkvolunteercollaborative.org", "communityservice.org", "retiredbrains.com/senior-living-resources/volunteering"]
]
class EVCResourceLinksTableViewController: UITableViewController {
    var module: String = ""
    var resLinks = [String]()

    override func viewDidLoad() {
        super.viewDidLoad()
        let insets: UIEdgeInsets? = UIEdgeInsetsMake(0, 0, (self.tabBarController?.tabBar.frame.height)!, 0)
        self.tableView.contentInset = insets!
        self.tableView.scrollIndicatorInsets = insets!
        self.edgesForExtendedLayout = .all
        self.reloadData()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        self.view.makeToastActivity(.center)
        self.reloadData()
        self.view.hideToastActivity()
    }

    func reloadData() {
        self.resLinks = RESOURCE_LINKS[self.module] ?? [String]()
        self.tableView.reloadData()
    }

    
// MARK: - Table view data source

override func numberOfSections(in tableView: UITableView) -> Int {
        // Return the number of sections.
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // Return the number of rows in the section.
        return self.resLinks.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: .default, reuseIdentifier: "CELL")
        if self.resLinks.isEmpty {
            cell.textLabel?.text = "No Links Found"
        }
        else {
            cell.textLabel?.text = self.resLinks[indexPath.row]
        }
        return cell
    }
// MARK: - Table view delegate
    // In a xib-based application, navigation from a table can be handled in -tableView:didSelectRowAtIndexPath:

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        if self.resLinks.count > 0 {
            if self.module == "Jobs" && indexPath.row == 1 {
                self.openURLinBrowser(self.resLinks[indexPath.row], useWWW: false)
            } else {
                self.openURLinBrowser(self.resLinks[indexPath.row])
            }
            
        }
    }

    func openURLinBrowser(_ url: String, useWWW: Bool = true) {
        let fullURL: String
        if useWWW {
            fullURL = "http://www.\(url)"
        } else {
            fullURL = "http://\(url)"
        }
        UIApplication.shared.open(URL(string: fullURL)!, options: [:], completionHandler: nil)
    }
}
